src/utils.rs

```diff
@@ -1,19 +1,23 @@
 // msbd returns the most significant different bit index between the two given
-// numbers. The index is 0-based.
+// numbers. The index is 0-based counting from LSB to MSB.
 // if u and v are equal, it returns an index higher than than the bitsize.
 // TODO: make that generic for xorable values ?
 pub fn msbd(u: usize, v: usize) -> usize {
     let bitsize = node_bitsize();
     let xor = u ^ v;
-    let mut idx = 0; // index of the different bit
-    while idx < bitsize {
-        let diff = xor >> (bitsize - 1 - idx);
-        if diff == 1 {
+    if xor == 0 {
+        return bitsize + 1;
+        // FIXME: The +1 seems unnecessary, for a 0-based index
```

Comment:
Sure we can remove it. It's just what I wrote in the comment before coding the line so I return something greater than the bitsize but it's the same if we return `bitsize` ^^